src/asm/parser.y

```diff
@@ -1214,8 +1220,18 @@ reloc_16bit_no_str : relocexpr_no_str {
 				warning(WARNING_TRUNCATION, "Expression must be 16-bit\n");
 			$$ = $1;
 		}
+		| inline_fragment { rpn_Symbol(&$$, $1); }
 ;
 
+inline_fragment	: T_2LBRACK {
+			out_PushInlineFragmentSection();
+			sprintf($<inlineFragmentName>$, "$%" PRIu32, inlineFragmentID++);
+			sym_AddLabel($<inlineFragmentName>$);
+		} asmfile T_2RBRACK {
+			out_PopSection();
+			strcpy($$, $<inlineFragmentName>2);
```

Comment:
Didn't `$$` get written above already?

Reply:
No, that was the mid-action rule's value.